static/app/components/stories/matrix.tsx

```diff
@@ -1,38 +1,54 @@
-import {type ElementType} from 'react';
-import {isValidElement} from 'react';
+import {ComponentProps, type ElementType} from 'react';
 import styled from '@emotion/styled';
+import first from 'lodash/first';
 
+import JSXProperty from 'sentry/components/stories/jsxProperty';
 import SizingWindow from 'sentry/components/stories/sizingWindow';
 import {space} from 'sentry/styles/space';
 
-interface Props {
-  component: ElementType;
-  propMatrix: Record<string, unknown[]>;
-  selectedProps: [string, string];
+interface Props<E extends ElementType> {
+  propMatrix: Partial<
+    Record<
+      keyof ComponentProps<E>,
+      keyof ComponentProps<E> extends string ? unknown[] : never
+    >
+  >;
```

Comment:
I think you want this to be 
```suggestion
  propMatrix: Partial<{
  	[Prop in keyof ComponentProps<E>]: Array<ComponentProps<E>[Prop]>
  }>;
```

That should give better typing